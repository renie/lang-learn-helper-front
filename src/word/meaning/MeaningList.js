import React from 'react'

const MeaningList = ({meanings, MeaningListItem}) => {
    return (
        <ul>
            {meanings && meanings.map(meaning => <MeaningListItem key={meaning.id} meaning={meaning}/>)}
        </ul>
    )
}

export default MeaningList
