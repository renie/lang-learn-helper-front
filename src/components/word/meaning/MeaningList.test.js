import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'

import MeaningList from './MeaningList'

const MeaningListItem = ({meaning}) => `${meaning.id} - ${meaning.wordClass}`

it('renders correctly empty', () => {
    const tree = renderer
                    .create(<MeaningList 
                                meanings={undefined}
                                MeaningListItem={MeaningListItem}/>)
                    .toJSON()

    expect(tree).toMatchSnapshot()
})

it('renders correctly empty', () => {
    const tree = renderer
                    .create(<MeaningList 
                                meanings={[]}
                                MeaningListItem={MeaningListItem}/>)
                    .toJSON()

    expect(tree).toMatchSnapshot()
})

it('renders correctly not empty', () => {
 
    const meanings = [{
            "id": 1,
            "wordClass": "verb",
            "example": "He used my pen.",
            "description": "Verb use in past tense."
        },
        {
            "id":2,
            "wordClass":"adjective",
            "example":"My pen is used.",
            "description":"Means the object is not new anymore."
    }]

    const tree = renderer
                    .create(<MeaningList 
                                meanings={meanings}
                                MeaningListItem={MeaningListItem}/>)
                    .toJSON()

  expect(tree).toMatchSnapshot()
})
