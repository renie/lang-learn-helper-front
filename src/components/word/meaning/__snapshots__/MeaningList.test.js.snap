// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly empty 1`] = `<ul />`;

exports[`renders correctly empty 2`] = `<ul />`;

exports[`renders correctly not empty 1`] = `
<ul>
  1 - verb
  2 - adjective
</ul>
`;
